doc: Update steps for adding support for a language (#15625)
sourcegraph/sourcegraph: doc/dev/how-to/add_support_for_a_language.md

## Syntax highlighting support

> To support syntax highlighting on blob and diff views:

Reviewer: ```suggestion
To support syntax highlighting on code files, search results, diff views, and more:
```

1. Follow the [directions](https://github.com/sourcegraph/syntect_server#adding-languages) to add a language to [syntect server](https://github.com/sourcegraph/syntect_server).

> 1. Update the [sourcegraph/syntect_server](https://sourcegraph.com/search?q=repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24+-file:%5C.md+sourcegraph/syntect_server:&patternType=literal) image tags in [sourcegraph/sourcegraph](https://github.com/sourcegraph/sourcegraph).

Reviewer: ```suggestion
```

I updated https://github.com/sourcegraph/syntect_server#adding-languages to include mention of this step.

1. Update the [SyntectLanguageMap](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@56a9eec78566499b108e1f869712865d90cc29cf/-/blob/internal/highlight/syntect_language_map.go#L5:5) in [sourcegraph/sourcegraph](https://github.com/sourcegraph/sourcegraph).

> To support syntax highlighting in markdown code fences (in hovers):

Reviewer: ```suggestion
To support syntax highlighting in hovers:
```

"markdown code fences" is confusing here because search results are also markdown code fences, but syntect handles that not highlight.js ( 😕 )

> 1. Update the [highlight.js contributions](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@e7ffd56b10e9bae004dfbb5d7d1c1accc93072fd/-/blob/client/shared/src/highlight/contributions.ts#L21) map in [sourcegraph/sourcegraph](https://github.com/sourcegraph/sourcegraph).

Reviewer 2: @slimsag Rijnard mentioned that you had thought about using syntect for all syntax highlighting, including hovers. Is that something we want to seriously consider and not include the highlightjs instructions?

Reviewer: Potentially, yeah, it's weird that hovers support a lesser set of languages than code files and can produce different results/bugs - but I don't think that should prevent documenting the current state here at all.
